tests/Hedgehog.Tests/GenTests.fs

```diff
@@ -18,3 +18,16 @@ let ``dateTime creates System.DateTime instances`` count =
     |> List.distinct
     |> List.length
     =! actual.Length
+
+[<Fact>]
+let ``unicode doesn't return any surrogate`` () =
+    let actual = Gen.sample 100 100000 Gen.unicode 
+    [] =! List.filter System.Char.IsSurrogate actual
+
+[<Theory>]
+[<InlineData(65534)>]
+[<InlineData(65535)>]
+let ``unicode doesn't return any noncharacter`` nonchar =
+    let isNoncharacter = (=) <| Operators.char nonchar
+    let actual = Gen.sample 100 100000 Gen.unicode
+    [] =! List.filter isNoncharacter actual
```

Comment:
@moodmosaic There is a deterministic approach to test this function? These two tests are a little slow. The test run time on my computer is about twice as slow as before. Is it OK? or it's better to remove the tests?

Reply:
Gen.sample looks fine, if you want just lower the *count* (and the *size*) to, perhaps, something like

```
Gen.sample 10 100
```

This should be ~50% faster although I haven't measured it. As long as the examples [run in 10 seconds or less](http://blog.ploeh.dk/2012/05/24/TDDtestsuitesshouldrunin10secondsorless/) we should be good to go.

Reply:
```
$ dotnet test tests/Hedgehog.Tests/Hedgehog.Tests.fsproj
```

```
Build started, please wait...
Build completed.

Test run for [..]Hedgehog.Tests.dll(.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 15.7.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

Total tests: 178. Passed: 178. Failed: 0. Skipped: 0.
Test Run Successful.
Test execution time: 9.0759 Seconds
```

So, the whole test suite runs in ~9 seconds.

---

```diff
diff --git a/tests/Hedgehog.Tests/GenTests.fs b/tests/Hedgehog.Tests/GenTests.fs
index 66b2776..ade363f 100644
--- a/tests/Hedgehog.Tests/GenTests.fs
+++ b/tests/Hedgehog.Tests/GenTests.fs
@@ -21,7 +21,7 @@ let ``dateTime creates System.DateTime instances`` count =

 [<Fact>]
 let ``unicode doesn't return any surrogate`` () =
-    let actual = Gen.sample 100 100000 Gen.unicode
+    let actual = Gen.sample 10 100 Gen.unicode
     [] =! List.filter System.Char.IsSurrogate actual

 [<Theory>]
@@ -29,5 +29,5 @@ let ``unicode doesn't return any surrogate`` () =
 [<InlineData(65535)>]
 let ``unicode doesn't return any noncharacter`` nonchar =
     let isNoncharacter = (=) <| Operators.char nonchar
-    let actual = Gen.sample 100 100000 Gen.unicode
-    [] =! List.filter isNoncharacter actual
\ No newline at end of file
+    let actual = Gen.sample 10 100 Gen.unicode
+    [] =! List.filter isNoncharacter actual
```

```
$ dotnet test tests/Hedgehog.Tests/Hedgehog.Tests.fsproj
```

```
Build started, please wait...
Build completed.

Test run for [..]Hedgehog.Tests.dll(.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 15.7.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

Total tests: 178. Passed: 178. Failed: 0. Skipped: 0.
Test Run Successful.
Test execution time: 5.0906 Seconds
```

Indeed, changing Gen.sample *size* to 10 and *count* to 100 makes those two tests to run faster but also, like you wrote, makes them weaker.

---

Then, let's leave those two as-is for now and we can come back later and tune them if needed.

Reply:
> Then, let's leave those two as-is for now and we can come back later and tune them if needed.

Go for it plz.